core/storage/user/gae_models.py

```diff
@@ -672,3 +673,85 @@ def construct_model_id(cls, user_id, skill_id):
             str. The model id corresponding to the given user and skill.
         """
         return '%s.%s' % (user_id, skill_id)
+
+
+class UserContributionScoringModel(base_models.BaseModel):
+    """Model for storing the scores of a user for various suggestions created by
+    the user. Users having scores above a particular threshold for a category
+    can review suggestions for that category.
+
+    The id for this model is of the form '{{score_category}}.{{user_id}}'.
+    """
+
+    # The user id of the user.
+    user_id = ndb.StringProperty(required=True, indexed=True)
+    # The category of suggestion to score the user on.
+    score_category = ndb.StringProperty(required=True, indexed=True)
+    # The score of the user for the above category of suggestions.
+    score = ndb.FloatProperty(required=True, indexed=True)
+
+    @classmethod
+    def get_all_scores_of_user(cls, user_id):
+        """Gets all scores for a given user.
+
+        Args:
+            user_id: str. The id of the user.
+
+        Returns:
+            list(UserContributionsScoringModel). All instances for the given
+                user.
+        """
+        return cls.get_all().filter(cls.user_id == user_id).fetch()
+
+    @classmethod
+    def get_all_users_with_score_above_minimum_for_category(
+            cls, score_category):
+        """Gets all instances which have score above the
+        MINIMUM_SCORE_REQUIRED_TO_REVIEW threshold for the given category.
+
+        Args:
+            score_category: str. The category being queried.
+
+        Returns:
+            list(UserContributionsScoringModel). All instances for the given
+                category with scores above MINIMUM_SCORE_REQUIRED_TO_REVIEW.
+        """
+        return cls.get_all().filter(
+            cls.score_category == score_category).filter(
+                cls.score >=
+                suggestion_models.MINIMUM_SCORE_REQUIRED_TO_REVIEW).fetch()
+
+    @classmethod
+    def get_instance_id(cls, user_id, score_category):
```

Comment:
Can this be made private (e.g. be prefixed with ``_``)?

Reply:
Done